libnavigation-base/src/test/java/com/mapbox/navigation/base/route/options/NavigationOptionsTest.kt

```diff
@@ -87,4 +89,66 @@ class NavigationOptionsTest {
         assertEquals(options.distanceFormatter, distanceFormatter)
         assertEquals(options.onboardRouterConfig, routerConfig)
     }
+
+    @Test
+    fun fromBuilderToBuilderShouldBeEqual() {
+        val options = NavigationOptions.Builder()
+            .isFromNavigationUi(false)
+            .navigatorPollingDelay(123435)
+            .timeFormatType(TWENTY_FOUR_HOURS)
+            .isFromNavigationUi(true)
+            .build()
+
+        val other = options.toBuilder().build()
+
+        assertEquals(options, other)
+    }
+
+    @Test
+    fun shouldBuildEqualOptions() {
+        val optionsLhs = NavigationOptions.Builder()
+            .isFromNavigationUi(false)
+            .navigatorPollingDelay(123435)
+            .timeFormatType(TWENTY_FOUR_HOURS)
+            .isFromNavigationUi(true)
+            .build()
+
+        val optionsRhs = NavigationOptions.Builder()
+            .isFromNavigationUi(false)
+            .navigatorPollingDelay(123435)
+            .timeFormatType(TWENTY_FOUR_HOURS)
+            .isFromNavigationUi(true)
+            .build()
+
+        assertTrue(optionsLhs == optionsRhs)
+        assertEquals(optionsLhs, optionsRhs)
+    }
+
+    @Test
+    fun shouldBuildNonEqualOptions() {
+        val optionsLhs = NavigationOptions.Builder()
+            .navigatorPollingDelay(123435)
+            .timeFormatType(TWENTY_FOUR_HOURS)
+            .isFromNavigationUi(true)
+            .build()
+
+        val optionsRhs = NavigationOptions.Builder()
+            .navigatorPollingDelay(123435)
+            .timeFormatType(TWENTY_FOUR_HOURS)
+            .isFromNavigationUi(false)
+            .build()
+
+        assertTrue(optionsLhs != optionsRhs)
+        assertNotEquals(optionsLhs, optionsRhs)
+    }
+
+    @Test
+    fun pepe() {
+        val pepe = NavigationOptions.Builder()
+            .navigatorPollingDelay(123435)
+        val juan = NavigationOptions.Builder()
+            .navigatorPollingDelay(123435)
+
+        assertEquals(pepe, juan)
```

Comment:
```diff
diff --git a/libnavigation-base/src/main/java/com/mapbox/navigation/base/options/NavigationOptions.kt b/libnavigation-base/src/main/java/com/mapbox/navigation/base/options/NavigationOptions.kt
index d9deaec95..5b0e80976 100644
--- a/libnavigation-base/src/main/java/com/mapbox/navigation/base/options/NavigationOptions.kt
+++ b/libnavigation-base/src/main/java/com/mapbox/navigation/base/options/NavigationOptions.kt
@@ -43,13 +43,14 @@ data class NavigationOptions constructor(
         .onboardRouterConfig(onboardRouterConfig)
         .isFromNavigationUi(isFromNavigationUi)
 
-    class Builder {
-        private var roundingIncrement: Int = ROUNDING_INCREMENT_FIFTY
-        private var timeFormatType: Int = NONE_SPECIFIED
-        private var navigatorPollingDelay: Long = DEFAULT_NAVIGATOR_POLLING_DELAY
-        private var distanceFormatter: DistanceFormatter? = null
-        private var onboardRouterConfig: MapboxOnboardRouterConfig? = null
+    data class Builder(
+        private var roundingIncrement: Int = ROUNDING_INCREMENT_FIFTY,
+        private var timeFormatType: Int = NONE_SPECIFIED,
+        private var navigatorPollingDelay: Long = DEFAULT_NAVIGATOR_POLLING_DELAY,
+        private var distanceFormatter: DistanceFormatter? = null,
+        private var onboardRouterConfig: MapboxOnboardRouterConfig? = null,
         private var isFromNavigationUi: Boolean = false
+    ) {
 
         fun roundingIncrement(roundingIncrement: Int) =
             apply { this.roundingIncrement = roundingIncrement }
```

Fixes it 😉

Reply:
We had a bit of offline discussion about this, I was confused what we were talking about breaking (thanks for showing @Guardiola31337).

This could be a use case for the builder if someone wants to  keep around a `Builder` and do a `.distinctUntilChanged` or something.

But I would argue against that use case, because this builder is made to configure initialization. We create configurations in bulk and build them. We really care about the built configuration equality. So my opinion is this isn't a use-case and still prefer the class version. Also would rather not maintain generating `equals()` functions because people forget to do it.

What do you guys think @LukasPaczos @Guardiola31337 ?

Reply:
We're getting the equals/hash for free by using the data class and it doesn't require any other maintenance in the builder's case. We're declaring all of the fields anyway.

Reply:
One thing that we were discussing was that if you use `data class` you're forced to pass in parameters (which sometimes could cause issues because of default parameters and their order https://github.com/mapbox/driver/pull/455/files#diff-a3f69a49c3844fe81065c75d953d60d1R130-R134) while if you use `class` it's more explicit `Foo.Builder().name(name)` and removes the possibility of messing up but then you lose `data class` benefits ¯\\\_(ツ)\_/¯ I don't have a strong opinion here both have pros and cons.

Reply:
Same for me tbh :man_shrugging:

Reply:
getting unused features for free, isn't really getting anything free (in regards to the free equals())

we may be making bigger configuration changes soon so we can revisit. let's consider our approaches when doing https://github.com/mapbox/mapbox-navigation-android/issues/2469